src/stats/scatter.ts

```diff
@@ -0,0 +1,26 @@
+import {
+  Table,
+  ScatterLayerConfig,
+  ScatterMappings,
+  ScatterScales,
+  SizedConfig,
+} from '../types'
+import {getFillScale} from '../utils/getFillScale'
+import {getSymbolScale} from '../utils/getSymbolScale'
+import {appendGroupCol} from '../utils/appendGroupCol'
+
+export const scatterStat = (
+  config: SizedConfig,
+  layer: ScatterLayerConfig
+): {table: Table; mappings: ScatterMappings; scales: ScatterScales} => {
+  const table = appendGroupCol(config.table, layer.fill)
```

Comment:
See my comment below about the `fill` / `scale` mappings. I think we might want to be appending a second group column here based off of `layer.fill`.

Reply:
will return to this as discussed ^^